NumLib/Extrapolation/LocalLinearLeastSquaresExtrapolator-impl.h

```diff
@@ -0,0 +1,133 @@
+#include <functional>
+
+#include <logog/include/logog.hpp>
+#include <Eigen/Core>
+
+#include "AssemblerLib/SerialExecutor.h"
+#include "MathLib/LinAlg/BLAS.h"
+#include "MathLib/LinAlg/MatrixVectorTraits.h"
+#include "NumLib/Function/Interpolation.h"
+#include "LocalLinearLeastSquaresExtrapolator.h"
+
+namespace NumLib
+{
+
+template<typename GlobalVector, typename PropertyTag, typename LocalAssembler>
+void
+LocalLinearLeastSquaresExtrapolator<GlobalVector, PropertyTag, LocalAssembler>::
+extrapolate(LocalAssemblers const& local_assemblers, PropertyTag const property)
+{
+    _nodal_values.setZero();
+
+    // counts the writes to each nodal value, i.e., the summands in order to
+    // compute the average afterwards
+    auto counts =
+        MathLib::MatrixVectorTraits<GlobalVector>::newInstance(_nodal_values);
```

Comment:
There is `LocalToGlobalIndexMap::getNumElementDOF()` but we need the other way around relation. Keep it for now...